### PR TITLE
Save settings

### PR DESCRIPTION
**Pull request by nickclaw**

### CODE REVIEW

File: src/com/uwecocar2/appdeployer/AppDeployerWindow.java [hunk 1]
```diff
@@ -33,13 +37,15 @@
  * Displays logs in the console.
  */
 public class AppDeployerWindow implements PrintsMessages {
-	private static final String DEFAULT_SOURCE_PATH = System.getProperty("user.home") + "/Documents/GitHub/UWCenterStack";
-	private static final String DEFAULT_PLAYBOOK_IP = "169.254.0.1";
-	private static final String DEFAULT_PLAYBOOK_PASSWORD = "playbook";
-	private static final String[] PLAYBOOK_PINS = {"501138E7", "502CEE27", "50303968"};
-	private static final String DEFAULT_TABLET_SKD = "/Developer/SDKs/Research In Motion/BlackBerry WebWorks SDK for TabletOS 2.2.0.5";
-	private static final String DEFAULT_ROOT_HTML = "srcs/t2c/coord.html";
+	private static String default_source_path;
```
Comment (MixMasterMitch) [inline src/com/uwecocar2/appdeployer/AppDeployerWindow.java:42]: These should no longer be static and should follow normal camel casing: `private String defualtSourcePath;`

File: src/com/uwecocar2/appdeployer/AppDeployerWindow.java [hunk 2]
```diff
@@ -33,13 +37,15 @@
...
+	private static String default_playbook_ip;
+	private static String default_playbook_password;
+	private static String[] playbook_pins;
+	private static int playbook_pins_index;
+	private static String default_tablet_sdk;
+	private static String default_root_html;
 	private static final String MAIN_WINDOW_TITLE = "Deploy App";
+	private static final String DEFAULT = "/Documents/GitHub/UWCenterStack\n169.254.0.1\nplaybook\n501138E7\n502CEE27\n50303968\n0\n/Developer/SDKs/Research In Motion/BlackBerry WebWorks SDK for TabletOS 2.2.0.5\nsrcs/t2c/coord.html";
```
Comment (MixMasterMitch) [inline src/com/uwecocar2/appdeployer/AppDeployerWindow.java:50]: I think that these should be stored in a better format than just a giant string. (See other comments on creating a new class for managing settings)

File: src/com/uwecocar2/appdeployer/AppDeployerWindow.java [hunk 3]
```diff
@@ -103,6 +106,15 @@ public void actionPerformed(ActionEvent event)
 				}
 			}
 		});
+		
+		//Setup deployButton
```
Comment (MixMasterMitch) [inline src/com/uwecocar2/appdeployer/AppDeployerWindow.java:172]: `//Setup defaultButton`

File: src/com/uwecocar2/appdeployer/AppDeployerWindow.java [hunk 4]
```diff
@@ -195,6 +207,96 @@ public void actionPerformed(ActionEvent event)
...
 
 	}
+	
+	/**
+	 * Reads text.txt for stored settings
+	 * @return String array of all settings|String array of default settings
+	 */
+	public String[] getSettings() {
+		try {
+			Scanner s = new Scanner(new File(System.getProperty("user.home") + "/Library/Application Support/CenterStack/settings"));
```
Comment (MixMasterMitch) [inline src/com/uwecocar2/appdeployer/AppDeployerWindow.java:280]: I think this file path should be stored in a final static variable.

**Comment by MixMasterMitch** [comment]
Ok! This all looks great with one exception: you are using one class for multiple different operations. I think that you should create two more classes DeploymentSettings and DeploymentSettingsManager. The DeploymentSettings class would contain all of the settings fields (playbookIP, skdPath, projectPath, etc.) and getter and setter methods. The DeploymentSettingsManager would have a getSettings, getDefualtSettings, and saveSettings method. The getSettings would use the default settings if the saved settings file is not found.

Additionally I think you should improve the settings text file format. I would suggest you go with something like:

```
playbookIP=123.456.0.1
playbookPins=abcdefg, asdfjk234, asdf0234
projectRoot=~/Documents/GitHub/UWCenterStack
...
```

which you would read in line by line and put into a map:

```
Map<String, String> settingsMap;
String[] parts = line.split("=");
settingsMap.put(parts[0], parts[1]);
```

and then you can put it into a DeploymentSettings object like this:

```
DeploymentSettings settings;
settings.setPlaybookIP(settingsMap.get("playbookIP"));
settings.setPlaybookPins(settingsMap.get("playbookPins").split(", "));
...
```

Sound good?

**Comment by nickclaw** [comment]
Sounds good to me, I probably wont have time to make any changes until this weekend though.

File: src/com/uwecocar2/appdeployer/AppDeployerWindow.java (hunk 2 above)
Comment (nickclaw) [inline src/com/uwecocar2/appdeployer/AppDeployerWindow.java:50]: Any other ideas on how to store it? Or do you mean keep it as a long string, but just in an out-of-the-way class...

**Pull request closed without merging**